ipatests/prci_definitions/prci_checker.py

```diff
@@ -0,0 +1,198 @@
+import click
+
+HINT = click.style("HINT: ", fg="bright_magenta")
+WARNING = click.style("WARNING: ", fg="yellow")
+ERROR = click.style("ERROR: ", fg="red")
+
+# Jobs specification file location
+JOBS_SPEC_YAML = "prci_jobs_spec.yaml"
+# Files to ignore on check
+IGNORE_FILES = {JOBS_SPEC_YAML, "temp_commit.yaml"}
+
+
+def load_yaml(path):
+    """Load YAML file into Python object."""
+    with open(path, "r") as file_data:
+        data = yaml.safe_load(file_data)
+        # data = yaml.load(file_data)
+    return data
+
+
+def check_jobs(filename, def_suite, jobs_spec):
+    """
+    Function that checks for a given definition file it has
+    a set of fields correctly set for each job, as defined in the job
+    specification file.
+
+    :param filename: file name of the definition file to be checked
+    :param def_suite: definition file as a dict object
+    :param jobs_spec: PRCI specification file containing correct definitions
+    """
+    current_spec = jobs_spec["prci_job_spec"].get(filename)
+    if current_spec is None:
+        click.echo(
+            WARNING
+            + f"{filename} file is not defined in the PRCI specification"
+            f" ({JOBS_SPEC_YAML}) file and could not be analyzed."
+        )
+        return
+
+    jobs_def = def_suite.get("jobs")
+    if jobs_def is None:
+        click.echo(
+            ERROR
+            + f"{filename} file doesn't have a jobs section following the"
+            " format."
+        )
+        return
+    click.echo("[File] " + filename + " [Jobs] " + str(len(jobs_def)))
+
+    try:
+        job_prefix = current_spec["job_prefix"]
+        requires = current_spec["requires"]
+        build_url = current_spec["build_url"]
+    except KeyError:
+        click.echo(ERROR + f"{JOBS_SPEC_YAML} has bad format.")
+        raise
+    build_job = job_prefix + "build"
+    build_args = jobs_def[build_job]["job"]["args"]
+    template = build_args["template"]
+    copr = build_args.get("copr")
+    update_packages = current_spec.get("update_packages")
+    selinux = current_spec.get("selinux_enforcing")
+
+    for job_name, params in jobs_def.items():
+        if job_name == build_job:  # Build job not analyzed
+            continue
+        args = params.get("job").get("args")
+        if not job_name.startswith(job_prefix):
+            click.echo(
+                ERROR
+                + f"{job_name} in {filename} file."
+                f"\nJob name must start with prefix {job_prefix}"
+            )
+        if params.get("requires") != requires:
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "requires" field must'
+                f" be defined as requires: {requires}"
+            )
+        if args.get("build_url") != build_url:
+            click.echo(
+                ERROR
+                + f'{job_name} in {filename} file.\nJob "build_url" arg field'
+                f" must be defined as build_url: {build_url}"
```

Comment:
On multiple places, it might better readable if the expected field value would be encapsulated in `` ` ` ``. E.g.: ``Job "update_packages" arg field must be defined as `update_packages: True` `` to spot easier what is the expected value and what is other error text.

Also what about having the general error format:
```
ERROR: In job ${JOBNAME}:
 Field: "${FIELDNAME}" should: `error text`
```

Alternativeluy:
```
Job field "${FIELDNAME}" must be defined as: `${EXPECTED}`
```

Atm in each error is repeated the file name. I'm not sure if it is needed. From my PoV, it makes more difficult to spot the job name and the file name is also mentioned in the [file] line.

Reply:
Refactored. I went with a generic format:
```
ERROR: In job 'fedora-latest/test_installation_TestInstallMaster':
  Job field "requires" should be defined as: 'requires: ['fedora-latest/build']'
```